### PR TITLE
Release/Fix unread count when moving file between nodes

### PR DESCRIPTION
**Pull request by samanehsan**
Purpose:
------------
When moving a file to a different component, the `node` field for all of its comments get updated to the destination node. This updated the `date_modified` field, which resulted in all of the comments appearing as if they were unread.

Requires: https://github.com/CenterForOpenScience/osf.io/pull/5254 (removes the auto-update of `date_modified`)

Changes
------------
Update the `user.comments_viewed_timestamp` so that `user.comments_viewed_timestamp[destination_node._id]['files'][root_target_id] == user.comments_viewed_timestamp[source_node._id]['files'][root_target_id]`

### CODE REVIEW

File: website/project/views/comment.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 
 @file_updated.connect
-def update_file_guid_referent(self, node, event_type, payload, user=None):
+def update_file_guid_referent(self, node, user, event_type, payload):
```
Comment (sloria) [inline website/project/views/comment.py]: Good call on making `user` a required argument :+1: .

**Comment by sloria** [comment]
Pass finished. Looks good to merge when the build passes. :octopus: 

As discussed in person, this change will be unnecessary if we change `comments_viewed_timestamp` to be keyed on GUID rather than node ID. Saman will send a follow-up PR implementing this refactor.

**Comment by samanehsan** [comment]
The follow-up PR is #5268

**Comment by samanehsan** [comment]
Replaced by #5268

**Pull request closed without merging**